### PR TITLE
Verify only uploaded (pre-crop) image size

### PR DESCRIPTION
**Pull request by misicnenad**
Ticket(s):
- https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1973

## Explanation of the solution
The idea is to only validate the original image that the user uploaded to `ImgEditor` and not the cropped/processed image (for explanation see [this comment](https://app.clickup.com/t/862jg6t1j?comment=90040006955728&threadedComment=90040006971841)).
To do that, we store the original (precropped) image and only focus on its validation (type, size etc.). If original image is valid, the cropped image will be valid for sure.

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
- connect Charity 1 wallet
- go to Edit Profile page http://localhost:4200/admin/1/edit-profile
- verify new tooltip appears under image upload components
- verify image upload works as before
- verify user is only ever able to upload images that have a size of `<= 1MB` (no matter the cropped image size)

### CODE REVIEW

File: src/components/ImgEditor/ImgEditor.tsx [hunk 1]
```diff
@@ -112,18 +112,22 @@ export default function ImgEditor<T extends FieldValues, K extends keyof T>(
...
-          Valid types are: JPG, JPEG, PNG and WEBP. Original uploaded image
-          should be less than 1MB in size.
-          <br />
-          {file
-            ? `Current (cropped) image size: ${humanize(
-                file.size / BYTES_IN_MB
-              )}MB.`
-            : ""}
+          Valid types are: JPG, JPEG, PNG and WEBP.{" "}
+          {maxSize ? (
```
Comment (misicnenad) [inline src/components/ImgEditor/ImgEditor.tsx:116]: No need to show the "max size" tooltip if there's no maximum size set.

File: src/components/ImgEditor/ImgEditor.tsx [hunk 2]
```diff
@@ -112,18 +112,22 @@ export default function ImgEditor<T extends FieldValues, K extends keyof T>(
...
+            <>
+              Image should be less than {maxSize / BYTES_IN_MB}MB in size.
+              <br />
+              {imgSize
```
Comment (misicnenad) [inline src/components/ImgEditor/ImgEditor.tsx:120]: Current image size is only relevant if there is a maximum image size. 
@SovereignAndrey we can show this nevertheless of course, this is just a proposal.

File: src/components/ImgEditor/types.ts [hunk 3]
```diff
@@ -13,4 +17,8 @@ export type Props<T extends FieldValues, K extends keyof T> = {
   accept: string[];
   classes?: Classes;
   aspect: [number, number];
+  /**
+   * Maximum image size in bytes
```
Comment (misicnenad) [inline src/components/ImgEditor/types.ts:21]: Adding JSDoc comments can be very helpful during development as it provided useful info in code editors (see image for on-hover info in VS Code).
image

We should make it a habit to add them, at least for fields/params whose purpose is not immediately obvious.

File: src/components/ImgEditor/types.ts [hunk 4]
```diff
@@ -13,4 +17,8 @@ export type Props<T extends FieldValues, K extends keyof T> = {
...
+   */
+  maxSize?: number;
```
Comment (misicnenad) [inline src/components/ImgEditor/types.ts:23]: Adding `maxSize` prop makes `ImgEditor` tooltip at the bottom reflect the actual max image size (currently was hardcoded)

File: src/components/ImgEditor/ImgEditor.tsx (hunk 2 above)
Comment (SovereignAndrey) [reply src/components/ImgEditor/ImgEditor.tsx:120]: I can't imaging a world where we don't have _some form_  of upper limit on file size uploads. Even if large, it'd still need to exist.

**Review by SovereignAndrey — APPROVED** [review]

**Review by ap-justin — APPROVED** [review]

**Pull request merged**